pyro/poutine/uncondition_messenger.py

```diff
@@ -19,6 +19,7 @@ def _pyro_sample(self, msg):
         if msg["is_observed"]:
             msg["is_observed"] = False
             msg["infer"]["was_observed"] = True
+            msg["infer"]["obs"] = msg["value"]
```

Comment:
@eb8680 I was able to get things working with this change to `uncondition`. Is it reasonable?

Reply:
Looks fine to me